virttest/utils_disk.py

```diff
@@ -659,39 +693,74 @@ def configure_empty_windows_disk(session, did, size, start="0M",
 
     :param session: session object to guest.
     :param did: disk index which show in 'diskpart list disk'.
-    :param size: partition size. e.g. 500M
+    :param size: the total size of partitions. e.g. 2G
     :param start: partition beginning at start. e.g. 0M
     :param n_partitions: the number of partitions on disk
     :param fstype: filesystem type for the disk.
     :param labeltype: label type for the disk.
     :param timeout: Timeout for cmd execution in seconds.
     :return a list: mount point list for all partitions.
     """
+
+    def _get_availabel_size():
+        """Get the free size of the disk."""
+        disk_script = '_'.join(("disk", ''.join(
+            random.sample(string.ascii_letters + string.digits, 4))))
+        cmd_list = "echo list disk >"
+        cmd_exit = "echo exit >>"
+        cmd_diskpart = "diskpart /s"
+        cmd_del = "del /f %s" % disk_script
+        cmd = (' %s && ' % disk_script).join((cmd_list, cmd_exit,
+                                              cmd_diskpart, cmd_del))
+        pattern = re.compile(r'Disk\s+%s\s+[A-Z]+\s+'
+                             r'\d+\s+[A-Z]+\s+'
+                             r'(\d+\s+[A-Z]+)' % did,
+                             re.I | re.M)
+        disk_info = session.cmd(cmd, timeout=timeout)
+        for info in disk_info.splitlines():
```

Comment:
disk_info was a string with multi lines already, and the pattern also have re.M, so I thinks no need 720 line here, just search pattern in disk_info is ok. like:
s = pattern.search(disk_info)
if s:
    return s.group(1).strip()

Reply:
thanks, have updated.